server/routes_test.go

```diff
@@ -233,6 +233,24 @@ func Test_Routes(t *testing.T) {
 				assert.Equal(t, expectedParams, params)
 			},
 		},
+		{
+			Name:   "Retrieve Model Handler OpenAI",
```

Comment:
This works, however what would be even better is some tests in `openai.go` that tests the middleware in isolation. Here's an example I had an LLM generate me for a middleware that changes formats (not the same although I think you get the idea)

```
package main

import (
    "net/http"
    "net/http/httptest"
    "testing"

    "github.com/gin-gonic/gin"
)

func TestFormatMiddleware(t *testing.T) {
    // Create a test router
    r := gin.New()
    r.Use(FormatMiddleware())
    r.GET("/test", func(c *gin.Context) {
        format := c.GetString("format")
        if format == "json" {
            c.JSON(http.StatusOK, gin.H{"message": "Success"})
        } else if format == "xml" {
            c.XML(http.StatusOK, gin.H{"message": "Success"})
        } else {
            c.String(http.StatusOK, "Success")
        }
    })

    // Define test cases
    tests := []struct {
        name           string
        queryParam     string
        expectedStatus int
        expectedBody   string
    }{
        {"JSON Format", "json", http.StatusOK, `{"message":"Success"}`},
        {"XML Format", "xml", http.StatusOK, `<map><message>Success</message></map>`},
        {"Default Format", "", http.StatusOK, "Success"},
    }

    for _, tt := range tests {
        t.Run(tt.name, func(t *testing.T) {
            req, _ := http.NewRequest("GET", "/test?format="+tt.queryParam, nil)
            w := httptest.NewRecorder()
            r.ServeHTTP(w, req)

            if w.Code != tt.expectedStatus {
                t.Errorf("Expected status %d, got %d", tt.expectedStatus, w.Code)
            }

            if w.Body.String() != tt.expectedBody {
                t.Errorf("Expected body %s, got %s", tt.expectedBody, w.Body.String())
            }
        })
    }
}
```